### ISSUE TITLE
tf.nn.conv2d on GPU with data_format='NHWC' gives corrupted result for specific shapes

### ISSUE THREAD

**Issue by SeguinBe** [tensorflow/tensorflow#16798]
Basically what the title says.

For an image of size (1096, 2449) EXACTLY, not 1097 or 1095 or 2448 or 2450 (but 2451 for some reason produces the same effect). The bottom part of the convolution result gets corrupted. It only affects convolution done on the GPU with the 'NHWC' data format.

Honestly, it feels more of a cuDNN bug than anything, but not sure where to post it otherwise.

**Important note : I am on Ubuntu 14.04 hence I can not try tensorflow 1.5 which needs CUDA 9 which needs 16.04. So my test is done on 1.4 with cuda8 and cuDNNv6.**

------------------------

### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 14.04
- **TensorFlow installed from (source or binary)**: pip install 1.4 tensorflow-gpu
- **TensorFlow version (use command below)**: v1.4.0-19-ga52c8d9 1.4.1
- **Python version**: 3.5
- **CUDA/cuDNN version**: cuda 8 cudnn 6
- **GPU model and memory**: TITAN X Pascal
- **Exact command to reproduce**:

```python
import tensorflow as tf
from tensorflow.contrib import layers
sess=tf.InteractiveSession()
h, w = 1096, 2449
input_img = tf.placeholder(tf.float32, shape=(None, h, w, 3))
filters = np.random.randn(1,1,3,2)
with tf.device('/cpu:0'):
    out_cpu = tf.nn.conv2d(input_img, filter=filters, strides=(1,1,1,1), padding='VALID')
with tf.device('/gpu:0'):
    out_gpu = tf.nn.conv2d(input_img, filter=filters, strides=(1,1,1,1), padding='VALID')
    # The following actually works if you manually transpose to use the NCHW data_format
    # tmp = tf.transpose(input_img, (0,3,1,2))
    # out_gpu = tf.nn.conv2d(tmp, filter=filters, strides=(1,1,1,1), padding='VALID', data_format='NCHW')
    # out_gpu = tf.transpose(out_gpu, (0,2,3,1))

out1, out2 = sess.run((out_cpu, out_gpu), feed_dict={
    input_img: np.random.randn(1, h, w, 3)
})
plt.imshow(np.linalg.norm(out1-out2 , axis=-1)[0])
plt.colorbar()
```
Plotting the difference between the CPU conv and the GPU conv :

<img width="359" alt="capture d ecran 2018-02-06 a 12 31 03" src="https://user-images.githubusercontent.com/7132817/35857501-a74bb66e-0b39-11e8-9edd-98e69eba5c48.png">

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @yzhwang: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @yzhwang: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @yzhwang: It has been 16 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @yzhwang: It has been 108 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by reedwm** [comment]
/CC @chsigg is this a known issue?

**Comment by SeguinBe** [comment]
I'd have to doublecheck again, but I think with the current versions of cuda and tf, I was not able to reproduce it.

**Comment by reedwm** [comment]
OK in that case I'll close the issue. If you determine this is still an issue, I'll reopen.

**Comment by SeguinBe** [comment]
Quick update, I actually tried again with Ubuntu 18.04, Tensorflow 1.7, Cuda 9.0, cuDNN 7.1.2 (all installed with `conda install tensorflow-gpu`) and the problem is actually still present.

**Comment by SeguinBe** [comment]
To be more precise:

- If I install with `conda install tensorflow-gpu=1.8.0`, conda cares of installing cudatoolkit=9.0 and cudnn=7.1.2. In that case, the bug seems to persist.

- If I have cuda 9.0 and cudnn 7 installed at the system level, and run `pip install tensorflow-gpu` to get 1.8.0, things work as they should. Same if I try the relevant code snippet in a collab notebook with GPU.

Long story short, I have no clue what is happening.

**Comment by reedwm** [comment]
/CC @chsigg can you look into this.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @chsigg: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @chsigg: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @chsigg: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @chsigg: It has been 63 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @chsigg: It has been 79 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @chsigg: It has been 94 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by chsigg** [comment]
Hi Seguin, I'm very sorry about the slow response time.
I agree this looks like a cuDNN bug, and should have been fixed in cuDNN 7.1.3.
